python/pyarrow/table.pxi

```diff
@@ -6259,6 +6259,56 @@ def concat_tables(tables, MemoryPool memory_pool=None, str promote_options="none
     return pyarrow_wrap_table(c_result_table)
 
 
+def concat_recordbatches(recordbatches, MemoryPool memory_pool=None):
```

Comment:
```suggestion
def concat_record_batches(recordbatches, MemoryPool memory_pool=None):
```

? (to be consistent with `record_batch(..)`)

Reply:
We could also go for `concat_batches` in the end the apis have `from_batches` and `to_batches`. The naming seems to be obvious enough that it won't confuse users if we just use "batches" and it would be more convenient to write than `concat_record_batches`.

Any opinion on this one?

Reply:
I would be in favour of `concat_batches` - it is shorter and as you mentioned, we already mostly use only batches elsewhere (also `iter_batches`, `scan_batches`).

Reply:
`concat_batches` indeed sounds good!